RiotSwiftUI/Modules/LocationSharing/StartLocationSharing/Coordinator/LocationSharingCoordinator.swift

```diff
@@ -125,45 +125,31 @@ final class LocationSharingCoordinator: Coordinator, Presentable {
     }
     
     private func shareStaticLocation(latitude: Double, longitude: Double, coordinateType: LocationSharingCoordinateType) {
-        locationSharingViewModel.startLoading()
-        
-        parameters.roomDataSource.sendLocation(withLatitude: latitude, longitude: longitude, description: nil, coordinateType: coordinateType.eventAssetType()) { [weak self] _ in
-            guard let self = self else { return }
-            
-            self.locationSharingViewModel.stopLoading()
-            self.completion?()
-        } failure: { [weak self] error in
-            guard let self = self else { return }
-            
+        parameters.roomDataSource.sendLocation(withLatitude: latitude, longitude: longitude, description: nil, coordinateType: coordinateType.eventAssetType()) { _ in
+        } failure: { error in
             MXLog.error("[LocationSharingCoordinator] Failed sharing location", context: error)
-            self.locationSharingViewModel.stopLoading(error: .locationSharingError)
         }
+        
+        self.completion?()
```

Comment:
I understand your changes but I'd call the `completion` block in the `success` and `failure` anyway for consistency.

Reply:
Can you explain why?
Doing so, wouldn't I break the api invariant expecting a completion block to be called just once? 🤔